### DISCUSSION TITLE
Can we have a tag index pages without automatic pages listing?

### DISCUSSION THREAD

**Discussion by NicolasMassart** [squidfunk/mkdocs-material#4168] [Ideas]
Hi,
I was wondering if we could have a tag index page that doesn't display pages with tags when we don't have the `[TAGS]` in the page.
Currently it always adds the page list even if we dont add this token.
My usecase is to have `tags_extra_files` and list these pages only on the `tags_file` page.
Using `tags_file` allows to have clickable tags that leads to this index page. But we should be able not to include `[TAGS]` if we don't want to.

basically it mean removing [lines 95 and 96 in the tags plugin](https://github.com/squidfunk/mkdocs-material/blob/47801173ed1474d168e3e6aba6d6bd5af8bb4df1/src/plugins/tags/plugin.py#L95):

```python
        if not "[TAGS]" in markdown:
            markdown += "\n[TAGS]"
```

I could create a PR, but want to make sure there's not issue in doing this.

Thanks.

**Comment by squidfunk** [comment]
Thanks for reporting.

> I was wondering if we could have a tag index page that doesn't display pages with tags when we don't have the [TAGS] in the page.

That sounds like a very edgey use case. If we would remove the automatic addition of `[TAGS]`, we would break existing configurations and make it less "just work". I'm not sure this is a good fit. Could you maybe achieve the same with a different setup? I'm not sure I fully understand the requirements.

**Reply by NicolasMassart** [reply]
I understand the compatibility issue. Thanks.

For the use case, it's because I have a lot of tagged pages and listing them in one strait list on the index page is too much and not very nice.
So I would like to use the tag index page as a landing page, where people land when they click a tag in a page. Having an index page is the only  way so far to have a clickable tag.
Then in this landing page I would have links to other tag specific pages populated by the tag_extra_file config.

But if we can't remove the tag automatic addition, let's get one step back in the discussion. What my issue is is that the main tag index page is very hard to read when you have all the tagged pages generated in the same list. Maybe we could improve the way it looks with some config, have some columns or show the list like a navigation tree? Today it lists pages in a basic list without mirroring the pages location. Maybe we could change a few things here and of course make this optional and not the default?

**Reply by squidfunk** [reply]
Thanks for diving into the details.

From what I understand is that you want to manually populate the tags index with links to the extra tags indexes, as the overall tags index is too big, but you still want to keep the linking functionality from a page to the index.

If we can think of a way to auto-populate it, it might be a great addition to the existing tags plugin. However, this would probably mean to somehow exclusively link tags to certain pages. Maybe we could use the existing tags mapping to adjust the links from a page to an extra tags index, skipping the overall tags index? You could then create a separate Markdown file, which is not managed by the tags plugin, linking to all extra tags pages, and the tags would also link to their specific pages. Since we would need a canonical destination to link a tag from a page to an index, we either would have to make sure there's a 1:1 mapping or we'd just say "use the first in order" as a fallback.

Would that work for you? Tags linking to `tags_extra_files` would mean that you could just create a meta-tags index outside of the tags plugin. If so, we could maybe make linking behavior configurable.
